"cargo check"  makes  "cargo test"  fail

repo is at 
30a9879c461ce18ff120e513a2f6a16d0f2f4b80

````
cargo check
[...]
cargo test
[...]
test [run-pass] run-pass/regressions.rs ... ok
test [run-pass] run-pass/used_underscore_binding_macro.rs ... FAILED
test [run-pass] run-pass/returns.rs ... ok
test [run-pass] run-pass/whitelist/conf_whitelisted.rs ... ok
test [run-pass] run-pass/single-match-else.rs ... ok
failures:
---- [run-pass] run-pass/used_underscore_binding_macro.rs stdout ----
error: compilation failed!
status: exit code: 101
command: "target/debug/clippy-driver" "tests/run-pass/used_underscore_binding_macro.rs" "-L" "/home/matthias/vcs/github/rust-clippy/target/debug/test_build_base" "--target=x86_64-unknown-linux-gnu" "-C" "prefer-dynamic" "-o" "/home/matthias/vcs/github/rust-clippy/target/debug/test_build_base/used_underscore_binding_macro.stage-id" "-L" "target/debug" "-L" "target/debug/deps" "-Dwarnings" "-L" "/home/matthias/vcs/github/rust-clippy/target/debug/test_build_base/used_underscore_binding_macro.stage-id.aux"
stdout:
------------------------------------------
------------------------------------------
stderr:
------------------------------------------
error[E0464]: multiple matching crates for `serde`
  --> tests/run-pass/used_underscore_binding_macro.rs:10:10
   |
10 | #[derive(Deserialize)]
   |          ^^^^^^^^^^^
   |
   = note: candidates:
           crate `serde`: /home/matthias/vcs/github/rust-clippy/target/debug/deps/libserde-13a2511189973f33.rlib
error[E0463]: can't find crate for `_serde`
  --> tests/run-pass/used_underscore_binding_macro.rs:10:10
   |
10 | #[derive(Deserialize)]
   |          ^^^^^^^^^^^ can't find crate
error: aborting due to 2 previous errors
Some errors occurred: E0463, E0464.
For more information about an error, try `rustc --explain E0463`.
------------------------------------------
thread '[run-pass] run-pass/used_underscore_binding_macro.rs' panicked at 'explicit panic', /home/matthias/.cargo/registry/src/github.com-1ecc6299db9ec823/compiletest_rs-0.3.11/src/runtest.rs:2544:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.
failures:
    [run-pass] run-pass/used_underscore_binding_macro.rs
test result: FAILED. 27 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out
test compile_test ... FAILED
failures:
---- compile_test stdout ----
thread 'compile_test' panicked at 'Some tests failed', /home/matthias/.cargo/registry/src/github.com-1ecc6299db9ec823/compiletest_rs-0.3.11/src/lib.rs:89:22
failures:
    compile_test
test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out
error: test failed, to rerun pass '--test compile-test'
````
If I just run  cargo clean ; cargo test though, there is no panic.

--- a/tests/compile-test.rs
+++ b/tests/compile-test.rs
@@ -75,7 +75,10 @@ fn config(mode: &str, dir: PathBuf) -> compiletest::Config {
 }
 
 fn run_mode(mode: &str, dir: PathBuf) {
-    compiletest::run_tests(&config(mode, dir));
+    let cfg = config(mode, dir);
+    // clean rmeta data, otherwise "cargo check; cargo test" fails (#2896)
+    cfg.clean_rmeta();
+    compiletest::run_tests(&cfg);
 }
 
 fn run_ui_toml_tests(config: &compiletest::Config, mut tests: Vec<test::TestDescAndFn>) -> Result<bool, io::Error> {

